extensions/amp-animation/0.1/amp-animation.js

```diff
@@ -73,11 +73,15 @@ export class AmpAnimation extends AMP.BaseElement {
 
   /** @override */
   buildCallback() {
+    const ampdoc = this.getAmpDoc();
+
     user().assert(isExperimentOn(this.win, TAG),
         `Experiment "${TAG}" is disabled.`);
 
     // TODO(dvoytenko): Remove once we support direct parent visibility.
-    user().assert(this.element.parentNode == this.element.ownerDocument.body,
+    user().assert(
+        this.element.parentNode == this.element.ownerDocument.body ||
+        this.element.parentNode == ampdoc.getBody(),
```

Comment:
Can't the previous be removed?

Reply:
Not yet. The first one covers FIE and the second one covers PWA shadow doc. The first one will be gone as soon as all FIE are on ampdocs.